SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialVirtualWorkerTranslator.cpp

```diff
@@ -74,3 +140,141 @@ void SpatialVirtualWorkerTranslator::WriteMappingToSchema(Schema_Object* Object)
 		SpatialGDK::AddStringToSchema(EntryObject, SpatialConstants::MAPPING_PHYSICAL_WORKER_NAME, Entry.Value);
 	}
 }
+
+// This method is called on the worker who is authoritative over the translation mapping. Based on the results of the
+// system entity query, assign the VirtualWorkerIds to the workers represented by the system entities.
+void SpatialVirtualWorkerTranslator::ConstructVirtualWorkerMappingFromQueryResponse(const Worker_EntityQueryResponseOp& Op)
+{
+	// The query response is an array of entities. Each of these represents a worker.
+	for (uint32_t i = 0; i < Op.result_count; ++i)
+	{
+		for (uint32_t j = 0; j < Op.results[i].component_count; j++)
```

Comment:
Could cache the result of Op.results[i] at the start of the first loop to be used:
Op.results[i].component_count
Op.results[i].components[j]

Reply:
Sure